artcommon/artcommonlib/konflux/konflux_db.py

```diff
@@ -61,100 +48,134 @@ def add_build(self, build: konflux_build_record.KonfluxBuildRecord):
         # Fill in missing record fields
         build.ingestion_time = datetime.now()
         build.schema_level = SCHEMA_LEVEL
-        query = f'INSERT INTO `{self.table_ref}` {self.column_names} VALUES '
-        values = (f"'{value}'" if isinstance(value, str) else str(value) for value in build.to_dict().values())
-        query += '(' + ', '.join(values) + ')'
-        self.logger.info('Executing query: %s', query)
-
-        # This is using the standard table API, which could possibly exceed BigQuery quotas
-        # See https://cloud.google.com/bigquery/quotas#load_job_per_table.long for details
-        # In case this happens, we can use the streaming api, e.g. self.client.insert_rows_json(self.table_ref, builds)
-        # This can lead to unconsistent results, as there might be delays from the time a record is inserted,
-        # and the time it is actually available for retrieval. If we can't live with this limitation, we should consider
-        # adopting the new BigQuery Storage API, safe but harder to implement:
-        # https://github.com/googleapis/python-bigquery-storage/blob/main/samples/snippets/append_rows_proto2.py
-        self.client.query(query).result()
+
+        # Execute query
+        values = list(build.to_dict().values())
+        self.bq_client.insert(self.column_names, values)
 
     async def add_builds(self, builds: typing.List[konflux_build_record.KonfluxBuildRecord]):
         """
-        Insert a list of Konflux build records
+        Insert a list of Konflux build records using a parallel async loop to enable concurrent queries.
         """
 
         loop = asyncio.get_event_loop()
         with concurrent.futures.ThreadPoolExecutor() as pool:
             await asyncio.gather(*(loop.run_in_executor(pool, self.add_build, build) for build in builds))
 
-    def search_builds_by_fields(self, names: list, values: list):
-        query = f'SELECT * FROM `{self.table_ref}` WHERE '
-        where_clauses = [f'`{field}` = "{value}"' for field, value in zip(names, values)]
-        query += ' AND'.join(where_clauses)
-
-        self.logger.info('Executing query: %s', query)
-        results = self.client.query(query).result()
-        self.logger.info('Found %s builds', results.total_rows)
-
-        for row in results:
-            yield konflux_build_record.from_result_row(row)
-
-    def search_builds_by_field(self, name, value):
+    def search_builds_by_fields(self, names: list, values: list, order_by: str = '', sorting: str = 'DESC',
```

Comment:
Wondering why you would would use a dict here?
```suggestion
    def search_builds_by_fields(self, where: Dict[str, Any] = None, order_by: str = '', sorting: str = 'DESC',
```
where `where is None` implies no constraint and `where["name"] is None` implies `name IS NULL` in SQL? Two lists creates the unnecessary possibilities of mismatched fields.

Reply:
Agreed, done in 1de6476e2a5f4604b26d8e293c5a465d7eed5dc1